### ISSUE TITLE
Need MUCH MORE Unicode support

### ISSUE THREAD

**Issue by mpratt14** [StephenCasella/SoundCloud-Playlist-Sync#8]
Hi all,

Very amazing little program here. However, I often download tracks from my soundcloud likes list where the title of the artist and/or song is in japanese or has many symbols. Over time I have noticed that none of my tracks or artist folders that should have a symbol. Most folder names end up with a "." or a "_" at the end of the name instead of a symbol. Song names in japanese like this one...

https://soundcloud.com/space-magic/workouttape
"ワークアウトテープ"

end up like this...
"ãƒãƒããããƒˆãƒãƒãƒ_High_Quality.mp3"

This often causes me to have no idea which track this is, or who wrote it, especially when there is no metadata downloaded with the track. I don't know if it is related to the Soundcloud API or not but I assume that it could be an easy fix if the program actually encodes the tracks being downloaded.

**Comment by erwinkramer** [comment]
I fixed this is my fork, try it out or wait for the acceptance of the pull request by @StephenCasella  .

**Comment by mpratt14** [comment]
Hi,

Thanks very much for your attempt at fixing this. I just downloaded your
branch and I'll see if I can integrate it into an old version i have
installed.

On Wed, Sep 2, 2015 at 7:12 AM, erwinkramer <notifications@github.com>
wrote:

> I fixed this is my fork, try it out or wait for the acceptance of the pull
> request by @StephenCasella <https://github.com/StephenCasella> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/StephenCasella/SoundCloud-Playlist-Sync/issues/8#issuecomment-137034384>
> .
>



-- 
Sincerely,

Michael Pratt <mpratt51@gmail.com>

**Comment by erwinkramer** [comment]
Hi Michael,

Please check out this release, i included the installer with the first .zip download located here: https://github.com/erwinkramer/SoundCloud-Playlist-Sync/releases/tag/r1.0.0.37.1

**Comment by mpratt14** [comment]
thanks, that is easier :P

**Comment by mpratt14** [comment]
Hi,

Thanks again for your work. It is working great with Japanese and Arabic text that I tested so far as well as unicode text like "ＣＩＴＹ". There are only a few small things left that I see that it does not translate appropriately.

Two tracks I tested had basic parenthesis "()" that did not translate, as well as another track with an & symbol. This is missing in both the file name and the Title in the metadata.

Also, the cloud symbol "☁" is also not translating, for example with the track here...
https://soundcloud.com/vskeytaku/fast

These are not very important to me but I thought I would let you know.

**Comment by erwinkramer** [comment]
Thanks for testing. I just recreated the problem and i'll try to come up with a solution if i can.

If you got some more feature requests i'll be more than happy to try and add them!

**Comment by erwinkramer** [comment]
I think i fixed this issue Michael!

There was still some leftover correcting in place, back when UTF-8 wasn't enabled by default. I removed the majority of this. Still needs some testing though. Stay tuned for the next build.

**Comment by mpratt14** [comment]
Nice. While you're at it, here is a playlist of tracks with unique song/artist titles starting with the 5 that I tested first.

https://soundcloud.com/datpratt/sets/test

**Comment by erwinkramer** [comment]
Thanks for your list. I see that there still could be some improvement if possible, however i'm pretty surprised that in some cases the soundcloud downloader has better support for rare characters than Google Chrome and Internet explorer. See image for example:

https://cloud.githubusercontent.com/assets/4255748/9682038/311e2f12-5303-11e5-88e1-4194f96aed95.png

I'll look in this again but it'll be more difficult to improve.

**Comment by mpratt14** [comment]
Yeah, that is common and has happened to me too.

I highly recommend getting Google's Noto Sans font and making that the default for Chrome.

Alternatively, just going here chrome://settings/fonts and changing the first font to Segoe UI (if it is not that already) will fix that because that is the same font that Windows is likely using and has a slightly better number of unicode characters.

Also at the bottom of chrome://settings/fonts make sure encoding is UTF-8 or better.

**Comment by erwinkramer** [comment]
Thanks for the info. My last build is up at: https://github.com/erwinkramer/SoundCloud-Playlist-Sync/releases/tag/r1.0.0.39

I was not able to completely fix the encoding unfortunately, this is beyond my skill set. Some characters work on certain songs, and those same characters sometimes don't work. This is a weird issue at the point (literally the save() function) where metadata of the song is written to the actual file by the Taglib library. 

I think this will be my final build for a while. Enjoy it. I hope some others continue development.

**Comment by mpratt14** [comment]
besides the metadata, can you fix the encoding for the file name or is that
the same issue?

**Comment by erwinkramer** [comment]
Not sure. I might look at it some time, but don't expect any progress though.

**Comment by mpratt14** [comment]
ok, thanks.

**Comment by mpratt14** [comment]
Hi again,

Unfortunately it looks like your latest build "abfda68" has a bug where it does not show download progress and downloads multiple copies with increasing file extenstion tags (.mp3 then .mp3.mp3 then .mp3.mp3.mp3 etc.).

See image
https://imgur.com/v6EJRcF

**Comment by mpratt14** [comment]
(I did not try your 0.38 build, but I know your 0.37.1 build was working)

**Comment by erwinkramer** [comment]
Hi @mpratt14 ,

New version is up with some nice fixes. https://github.com/erwinkramer/SoundCloud-Playlist-Sync/releases

**Comment by mpratt14** [comment]
Hi,

Your recent build is working great. Best one yet man! :smile: However, here are 2 things that I noticed. They are not very important but might as well address...

I had to open it directly with the .exe because it would not install. here is log: https://www.dropbox.com/s/vkg0i331l4bxb4w/ICZKA0LD.log?dl=0

Also for some tracks, instead of artist name being under the tag Album Artist it is sometimes under Contributing Artists. It does not matter to me that the metadata not perfect, but whenever the tag for Contributing Artists gets written it appears to be the same lack of unicode support.
https://www.dropbox.com/s/40t21d38hjvhbuq/Screenshot%202015-09-05%2011.19.48%202.png
